external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -363,6 +369,7 @@ private boolean emitTupleIfNotEmitted(ConsumerRecord<K, V> record) {
                         } else {
                             collector.emit(tuple, msgId);
                         }
+                        tupleListener.onEmit(tuple, msgId);
```

Comment:
Is it intended that this isn't called in auto commit mode? I would understand why we wouldn't want to call it there, but I think we should maybe add a note to the interface so it's obvious to users.